src/nano_contracts/header.ts

```diff
@@ -152,8 +156,7 @@ class NanoContractHeader extends Header {
     let headerId;
     [headerId, buf] = [buf.subarray(0, 1), buf.subarray(1)];
 
-    const headerIdHex = headerId.toString('hex');
-    if (headerIdHex !== VertexHeaderId.NANO_HEADER) {
+    if (getVertexHeaderIdFromBuffer(headerId) !== VertexHeaderId.NANO_HEADER) {
       throw new Error('Invalid vertex header id for nano header.');
     }
```

Comment:
I think this can be even simpler.

```ts
    if (getVertexHeaderIdFromBuffer(buf) !== VertexHeaderId.NANO_HEADER) {
      throw new Error('Invalid vertex header id for nano header.');
    }
    buf = buf.subarray(1);
```

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/842/commits/582cfb4fc9874e358e2d734837b40dbc0780b156